worker/api.go

```diff
@@ -152,22 +154,56 @@ type SlabsUploadRequest struct {
 	CurrentHeight uint64     `json:"currentHeight"`
 }
 
+// SlabsUploadResponse is the response type for the /slabs/upload endpoint.
+type SlabsUploadResponse struct {
+	Slab     slab.Slab         `json:"slab"`
+	Metadata []HostInteraction `json:"metadata"`
+}
+
 // SlabsDownloadRequest is the request type for the /slabs/download endpoint.
 type SlabsDownloadRequest struct {
 	Slab      slab.Slice `json:"slab"`
 	Contracts []Contract `json:"contracts"`
 }
 
+// SlabsDownloadResponse is the response type for the /slabs/download endpoint.
+type SlabsDownloadResponse struct {
+	Data     []byte            `json:"data"`
+	Metadata []HostInteraction `json:"metadata"`
+}
+
 // SlabsDeleteRequest is the request type for the /slabs/delete endpoint.
 type SlabsDeleteRequest struct {
 	Slabs     []slab.Slab `json:"slabs"`
 	Contracts []Contract  `json:"contracts"`
 }
 
+// SlabsDeleteResponse is the response type for the /slabs/delete endpoint.
+type SlabsDeleteResponse struct {
+	Metadata []HostInteraction `json:"metadata"`
+}
+
 // SlabsMigrateRequest is the request type for the /slabs/migrate endpoint.
 type SlabsMigrateRequest struct {
 	Slab          slab.Slab  `json:"slab"`
 	From          []Contract `json:"from"`
 	To            []Contract `json:"to"`
 	CurrentHeight uint64     `json:"currentHeight"`
 }
+
+// SlabsMigrateResponse is the response type for the /slabs/migrate endpoint.
+type SlabsMigrateResponse struct {
+	Slab     slab.Slab         `json:"slab"`
+	Metadata []HostInteraction `json:"metadata"`
+}
+
+// A HostInteraction contains information about an interaction with a host.
+// These objects eventually end up as host interactions in the host
+// database.
+type HostInteraction struct {
+	Timestamp int64           `json:"timestamp"`
+	Type      string          `json:"type"`
+	HostKey   PublicKey       `json:"hostKey"`
+	Metadata  json.RawMessage `json:"metadata,omitempty"`
```

Comment:
the existing type calls this field `Result`, which I think is slightly better than `Metadata`

Reply:
I personally don't really like `Result` but I've updated it.